### PR TITLE
Fix memory leaks

### PR DESCRIPTION
**Pull request by cs278**
Fix memory leak identified by valgrind, also sets up CI job running valgrind.

### CODE REVIEW

**Review by la5942 — APPROVED** [review]
LGTM :+1:

File: .github/workflows/valgrind.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  build:
+    name: PHP ${{ matrix.php-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version:
+          - '5.6.40'
+          - '7.4.19'
+          - '8.0.7'
```
Comment (Jamesking56) [inline .github/workflows/valgrind.yml:17]: Very specific versions?

Comment (cs278) [reply .github/workflows/valgrind.yml:17]: They have to be unless there is a URL that gets the latest PHP 8.0 release etc.

**Review by deangray82 — APPROVED** [review]

File: .github/workflows/valgrind.yml (hunk 1 above)
Comment (cs278) [reply .github/workflows/valgrind.yml:17]: Couldn't find an alternative.

**Pull request merged**